feat: add `reset` to mouse

jrandolf-google opened the pull request:
This PR gives users the ability to reset their mouse without keeping track of their button presses and position.

OrKoN commented on packages/puppeteer-core/src/common/Input.ts:
```diff
@@ -598,6 +598,33 @@ export class Mouse {
+    if (this.#state.buttons & MouseButtonFlag.Forward) {
+      actions.push(this.up({button: MouseButton.Forward}));
+    }
+    if (this.#state.buttons & MouseButtonFlag.Back) {
```
want to convert it to an array + iteration? should be easier to read and less chance for copy paste errors.

jrandolf replied:
MouseButtonFlag is a const enum, so this isn't possible.

jrandolf replied:
I can alternatively create an array manually.

OrKoN replied:
yeah, I meant the latter option.

OrKoN replied:
i.e.,

```
const flags = [MouseButtonFlag.Left, ...];
for (const button of flags) {
   if (this.#state.buttons & button) {
      actions.push(this.up({button}));
    }
}
```

OrKoN reviewed (changes requested):
Let's add a test for this method.

Lightning00Blade commented on packages/puppeteer-core/src/common/Input.ts:
```diff
@@ -598,6 +598,33 @@ export class Mouse {
   }
 
+  /**
+   * Resets the mouse to the default state. No buttons pressed; position at
```
I think we should make it clear from the comment that we call `up` and `move`.
Something like finishes the started actions or similar.

jrandolf replied:
Those are implementation details. I'm not sure if we want to expose them.

Lightning00Blade replied:
I think we should as they convey that the appropriate events will be send on the page side.

OrKoN replied:
I agree with Nik. Because we have also an option to reset without sending the page commands, so we need to clarify that the events will be dispatched. Perhaps: `Moves the mouse to 0,0 position and un-presses all pressed buttons` or smth like that.

jrandolf committed:
feat: add `reset` to mouse

This PR gives users the ability to reset their mouse without keeping track of their button presses and position.

OrKoN commented on test/TestExpectations.json:
```diff
@@ -1457,6 +1457,12 @@
+    "testIdPattern": "[mouse.spec] Mouse should reset properly",
+    "platforms": ["darwin", "linux", "win32"],
+    "parameters": ["cdp", "firefox"],
+    "expectations": ["FAIL"]
```
```suggestion
    "expectations": ["SKIP"]
```

OrKoN committed:
chore: udate test/TestExpectations.json
